From 5cf264a4de9ec531a25ffee06f7de134283e7f98 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:23 -0400
Subject: [ethernet] i40e: remove support for vf unicast promiscuous mode

Message-id: <1409658042-22689-263-git-send-email-sassmann@redhat.com>
Patchwork-id: 90462
O-Subject: [RHEL7.1 PATCH 262/281] i40e: remove support for vf unicast promiscuous mode
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Ashish Shah <ashish.n.shah@intel.com>

Remove the ability of a VF to set unicast promiscuous mode.
Considered to be a security risk to allow VFs to receive traffic
intended for other VFs so don't allow it, simply ignore the flag.

Also fix it to send the correct seid to aq for multicast promiscuous set.

Change-ID: Icb9c49a281a8e9d3aeebf991ef1533ac82b84b14
Signed-off-by: Ashish Shah <ashish.n.shah@intel.com>
Tested-by: Jim Young  <jamesx.m.young@intel.com>
Signed-off-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 89cb86c3b2a7ec3326dc7ffe5cfb850818f78ea0)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
index 0087827..d688641 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
@@ -1163,8 +1163,8 @@ static int i40e_vc_config_promiscuous_mode_msg(struct i40e_vf *vf,
      (struct i40e_virtchnl_promisc_info *)msg;
  struct i40e_pf *pf = vf->pf;
  struct i40e_hw *hw = &pf->hw;
+ struct i40e_vsi *vsi;
  bool allmulti = false;
- bool promisc = false;
  i40e_status aq_ret;
 
  if (!test_bit(I40E_VF_STAT_ACTIVE, &vf->vf_states) ||
@@ -1174,17 +1174,10 @@ static int i40e_vc_config_promiscuous_mode_msg(struct i40e_vf *vf,
   aq_ret = I40E_ERR_PARAM;
   goto error_param;
  }
-
- if (info->flags & I40E_FLAG_VF_UNICAST_PROMISC)
-  promisc = true;
- aq_ret = i40e_aq_set_vsi_unicast_promiscuous(hw, info->vsi_id,
-           promisc, NULL);
- if (aq_ret)
-  goto error_param;
-
+ vsi = pf->vsi[info->vsi_id];
  if (info->flags & I40E_FLAG_VF_MULTICAST_PROMISC)
   allmulti = true;
- aq_ret = i40e_aq_set_vsi_multicast_promiscuous(hw, info->vsi_id,
+ aq_ret = i40e_aq_set_vsi_multicast_promiscuous(hw, vsi->seid,
              allmulti, NULL);
 
 error_param:
-- 
1.7.1